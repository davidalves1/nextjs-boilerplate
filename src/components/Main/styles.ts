import styled from 'styled-components';

export const Wrapper = styled.main`
  display: grid;
  place-content: center;
  place-items: center;
  background-color: #efefef;
  color: #333;
  width: 100vw;
  height: 100vh;
`;

export const Title = styled.h1`
  font-size: 4rem;
  font-weight: 200;
  margin-bottom: 1rem;
`;

export const Description = styled.h2`
  font-size: 2.5rem;
  font-weight: 200;
`;
