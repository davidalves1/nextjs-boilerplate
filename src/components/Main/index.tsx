import * as S from './styles';

const Main = () => (
  <S.Wrapper>
    <S.Title>NextJS App</S.Title>
    <S.Description>My Awesome Web App</S.Description>
  </S.Wrapper>
);

export default Main;
