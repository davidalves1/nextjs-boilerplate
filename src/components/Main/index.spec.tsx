import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    expect(screen.getByRole('heading', { name: /nextjs app/i })).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: /my awesome web app/i })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render ther colors correctly', () => {
    const { container } = render(<Main />);

    // How to test styles
    expect(container.firstChild).toHaveStyle({ 'background-color': '#efefef' });
  });
});
