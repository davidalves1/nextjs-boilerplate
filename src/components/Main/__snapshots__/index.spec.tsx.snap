// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<div>
  <main>
    <h1>
      React Avan√ßado
    </h1>
  </main>
</div>
`;
